Library/Homebrew/formula.rb

```diff
@@ -1510,7 +1508,20 @@ def eligible_kegs_for_cleanup
           end
         end
       end
-    elsif installed_prefixes.any? && !pinned?
+    elsif installed_prefixes.length > 1
+      if linked_keg.exist?
+        # If the most recent available version is not installed, cleanup
+        # only those older than the currently-linked one.
+        linked_version = PkgVersion.parse(linked_keg.resolved_path.basename.to_s)
```

Comment:
Should I just use `Keg.new(...)` (or something) here for maybe-slightly-clearer semantics? Or something else? Or is this fine?